### PR TITLE
implement closestPair algorithm and tests

### PR DESCRIPTION
**Pull request by nodatall**
> Closest pair is the ultimate sophistication.
~ Leonardo Da Vinci

I think he's been misquoted.
-Monica

### CODE REVIEW

**Comment by Moniarchy** [comment]
This looks good to me and also it might benefit from having someone else take a look as well, so I am going to put it back in the prrr stack.

**Review by jason00111 — COMMENTED** [review]
delete this comment pls

File: src/closestPair.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+const closestPair = array => {
+  if ( array.length < 2 ) {
+    return Infinity
+  } else {
+    let minimumDistance = distance( array[0], array[1] ),
+        minimumPoints = array.slice( 0, 2 ),
```
Comment (jrobcodes) [inline src/closestPair.js]: Not a fan of this indentation, it introduces a maintenance requirement to support formatting.  I would consider declaring separately, or indenting with the standard 2 space indentation.

File: src/closestPair.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    return { pair: minimumPoints, distance: minimumDistance }
+  }
+
+}
+
+const distance = (point1, point2) => {
+  const distanceX = Math.abs(point1[0] - point2[0]),
+        distanceY = Math.abs(point1[1] - point2[1])
```
Comment (jrobcodes) [inline src/closestPair.js]: Formatting

File: test/closestPair_test.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      expect(isFinite(closestPair(points))).to.equal(false)
+    })
+
+    it('returns the first two pairs if only two are given', () => {
+      const points = [ [1,2],[2,4] ]
+      expect(closestPair(points))
+        .to.deep.equal({ pair: [ [1,2],[2,4] ], distance: 2.23606797749979 })
+    })
+
+    it('should return the pair of points the least far away from one another', () => {
```
Comment (jrobcodes) [inline test/closestPair_test.js]: The phrasing is a little jarring - perhaps, "it should return the pair of points closest to each other"

**Pull request merged**